.github/workflows/create-release.yml

```diff
@@ -0,0 +1,84 @@
+name: "Create a release"
+
+on:
+  workflow_call:
+    inputs:
+      ref:
+        description: "Ref to be tagged"
+        required: true
+        type: string
+        default: master
+      tag:
+        description: "Tag for new version (v1.23.4)"
+        required: true
+        type: string
+      base_tag:
+        description: "Base tag to generate commit list for release notes"
+        required: false
+        type: string
+    secrets:
+      TEMPORAL_CICD_APP_ID:
+        required: true
+      TEMPORAL_CICD_PRIVATE_KEY:
+        required: true
+
+jobs:
+  create-release:
+    name: "Create a release"
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Generate token
+        id: generate_token
+        uses: actions/create-github-app-token@v1
+        with:
+          app-id: ${{ secrets.TEMPORAL_CICD_APP_ID }}
+          private-key: ${{ secrets.TEMPORAL_CICD_PRIVATE_KEY }}
+          owner: ${{ github.repository_owner }}
+
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          repository: temporalio/api-go
+          ref: ${{ github.event.inputs.ref }}
+          token: ${{ steps.generate_token.outputs.token }}
+          persist-credentials: true
+          fetch-depth: 0
+          fetch-tags: true
+          submodules: true
+
+      - name: Validate input
```

Comment:
We should also validate that the version given here matches the `api` submodule version. I think we should verify that there is a release or maybe a draft referencing the `api` submodule commit with the same version given as input to this workflow.

Reply:
Done, I also added the same validation in the `api` workflow before the release drafts are created.